### PR TITLE
Migrate FieldValue to Protobuf

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This PR removes the FieldValue classes and updates the client to directly use Protobuf messages internally. It aligns the iOS client with the Web and Android client.

CPU time benchmark On iPhone 8:

<google-sheets-html-origin>
Benchmark Name | Branch (us) | Master (us) | Speedup (x)
-- | -- | -- | --
BM_QueryIndexFree/0/1 | 7.68 | 18.6 | 2.42
BM_QueryIndexFree/1/1 | 12.3 | 27.7 | 2.25
BM_QueryIndexFree/1/10 | 13 | 28.7 | 2.21
BM_QueryIndexFree/10/10 | 22.4 | 25.9 | 1.16
BM_QueryIndexFree/1/100 | 11.9 | 29.2 | 2.45
BM_QueryIndexFree/100/100 | 8.87 | 24.3 | 2.74
BM_QueryIndexFree/1/1000 | 18.7 | 29 | 1.55
BM_QueryIndexFree/10/1000 | 17.8 | 28.9 | 1.62
BM_QueryIndexFree/100/1000 | 9.89 | 36.9 | 3.73
BM_QueryIndexFree/1000/1000 | 21.1 | 46.4 | 2.20
BM_QueryMatching/0/1 | 12.8 | 40.5 | 3.16
BM_QueryMatching/1/1 | 18.7 | 41.2 | 2.20
BM_QueryMatching/1/10 | 35.3 | 45.9 | 1.30
BM_QueryMatching/10/10 | 53.4 | 59.4 | 1.11
BM_QueryMatching/1/100 | 63.4 | 108 | 1.70
BM_QueryMatching/100/100 | 44.5 | 127 | 2.85
BM_QueryMatching/1/1000 | 60.3 | 251 | 4.16
BM_QueryMatching/10/1000 | 73.3 | 181 | 2.47
BM_QueryMatching/100/1000 | 24.7 | 149 | 6.03
BM_QueryMatching/1000/1000 | 27.8 | 150 | 5.40
BM_QueryAll/1 | 21 | 32.2 | 1.53
BM_QueryAll/10 | 24.5 | 34.5 | 1.41
BM_QueryAll/100 | 9.41 | 37.9 | 4.03
BM_QueryAll/1000 | 17.3 | Out of memory | Inf

</google-sheets-html-origin>

Full results (internal): go/fieldvalue-benchmark

XCode's Leak Detector does not detect any leaks with this PR when used in FriendlyEats.

### CODE REVIEW

**Comment by schmidt-sebastian** [comment]
I shared a trace with the remaining leaks (excluding spec tests). I don't think there are any more relevant leaks from this PR.

**Comment by schmidt-sebastian** [comment]
@var-const Verified that ASAN tests still pass and added more performance numbers to go/fieldvalue-benchmark

File: Firestore/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
+# Unreleased
+- [changed] Internal refactor to improve serialization performance.
+
 # v8.4.0
 - [fixed] Fixed handling of Unicode characters in log and assertion messages
-  (#8372).
```
Comment (var-const) [inline Firestore/CHANGELOG.md:3]: Is deleting this line deliberate?

Comment (schmidt-sebastian) [reply Firestore/CHANGELOG.md:3]: Fixed

**Review by var-const — APPROVED** [review]

**Pull request merged**